packages/next/src/build/webpack-config.ts

```diff
@@ -1770,7 +1775,8 @@ export default async function getBaseWebpackConfig(
           dev,
         }),
       (isClient || isEdgeServer) && new DropClientPage(),
-      (isNodeServer || (config.experimental.flyingShuttle && isEdgeServer)) &&
+      (isNodeServer ||
+        ((flyingShuttle || isStoreOnlyFlyingShuttle) && isEdgeServer)) &&
```

Comment:
I think we can remove the `isStoreOnlyFlyingShuttle` check here since if `flyingShuttle` is true then something is set